src/components/media-views.js

```diff
@@ -286,12 +288,34 @@ AFRAME.registerComponent("media-video", {
   play() {
     this.el.addEventListener("grab-start", this._grabStart);
     this.el.addEventListener("grab-end", this._grabEnd);
+    this.seekForwardButton.addEventListener("grab-start", this.seekForward);
+    this.seekBackButton.addEventListener("grab-start", this.seekBack);
+    this.seekForwardButton.object3D.visible = true;
+    this.seekBackButton.object3D.visible = true;
   },
 
   // aframe component pause, unrelated to video
   pause() {
     this.el.removeEventListener("grab-start", this._grabStart);
     this.el.removeEventListener("grab-end", this._grabEnd);
+    this.seekForwardButton.removeEventListener("grab-start", this.seekForward);
+    this.seekBackButton.removeEventListener("grab-start", this.seekBack);
+    this.seekForwardButton.object3D.visible = false;
+    this.seekBackButton.object3D.visible = false;
+  },
+
+  seekForward() {
+    if (NAF.utils.isMine(this.networkedEl) || NAF.utils.takeOwnership(this.networkedEl)) {
```

Comment:
How does this work when you aren't the owner? takeOwnership is async right? I guess you sync the currentTime property so the other users will get that update?

Reply:
takeOwnership is not actually async, you know if you are going to become the owner based on the last owner timestamp. You of course may lose ownership if someone else has a message in flight that has a newer owner time than yours, but that acts just like any other form of losing ownership. The only time it would ever return false is if you already own it (we already check for that) or if you somehow got a lastOwnerTime that is mroe recent than your current server time.